metrics/sinks/gcm/gcm.go

```diff
@@ -54,49 +50,66 @@ type gcmSink struct {
 	registered   bool
 	project      string
 	metricFilter MetricFilter
-	gcmService   *gcm.Service
+	gcmService   *v3.Service
 }
 
 func (sink *gcmSink) Name() string {
 	return "GCM Sink"
 }
 
-func getReq() *gcm.WriteTimeseriesRequest {
-	return &gcm.WriteTimeseriesRequest{Timeseries: make([]*gcm.TimeseriesPoint, 0)}
+func getReq() *v3.CreateTimeSeriesRequest {
+	return &v3.CreateTimeSeriesRequest{TimeSeries: make([]*v3.TimeSeries, 0)}
 }
 
-func fullLabelName(name string) string {
-	// handle correctly GCE specific labels
-	if !strings.Contains(name, "compute.googleapis.com") {
-		return fmt.Sprintf("%s/%s/label/%s", customApiPrefix, metricDomain, name)
-	}
-	return name
+func fullMetricName(project string, name string) string {
+	return fmt.Sprintf("projects/%s/metricDescriptors/%s/%s/%s", project, customApiPrefix, metricDomain, name)
 }
 
-func fullMetricName(name string) string {
+func fullMetricType(name string) string {
```

Comment:
This is as intended. There's a fullMetricName function as well (line 64). In Stackdriver v3, custom metric name and type are not the same (see https://cloud.google.com/monitoring/custom-metrics/creating-metrics, "Custom Metric Names").